diff --git a/dist/metro/index.cjs b/dist/metro/index.cjs
--- a/dist/metro/index.cjs
+++ b/dist/metro/index.cjs
@@ -146,7 +146,7 @@ const withUniwindConfig = (config$1, uniwindConfig) => {
     transformerPath: require.resolve("./transformer.cjs"),
     transformer: {
       ...config$1.transformer,
-      uniwind: bundlerConfig.toMetroConfig()
+      uniwind: { ...bundlerConfig.toMetroConfig(), isExpoProject: config$1.transformerPath?.includes("@expo/metro-config") }
     },
     resolver: {
       ...config$1.resolver,
diff --git a/dist/metro/transformer.cjs b/dist/metro/transformer.cjs
--- a/dist/metro/transformer.cjs
+++ b/dist/metro/transformer.cjs
@@ -1701,17 +1701,19 @@ const compileCSS = async (bundlerConfig) => {
 
 const cssArtifactPath = path__default.resolve(__dirname, "../../uniwind.css");
 let worker;
-try {
-  try {
-    const { unstable_transformerPath } = require("@expo/metro-config");
-    worker = require(unstable_transformerPath);
-  } catch {
-    worker = require("@expo/metro-config/build/transform-worker/transform-worker.js");
-  }
-} catch {
-  worker = require("metro-transform-worker");
-}
 const transform = async (config$1, projectRoot, filePath, data, options) => {
+  if (!worker) {
+    if (config$1.uniwind?.isExpoProject) {
+      try {
+        const { unstable_transformerPath } = require("@expo/metro-config");
+        worker = require(unstable_transformerPath);
+      } catch {
+        worker = require("@expo/metro-config/build/transform-worker/transform-worker.js");
+      }
+    } else {
+      worker = require("metro-transform-worker");
+    }
+  }
   const isCss = options.type !== "asset" && path__default.join(process.cwd(), config$1.uniwind.cssEntryFile) === path__default.join(projectRoot, filePath);
   if (filePath.endsWith("/components/web/metro-injected.js")) {
     const bundlerConfig2 = config.UniwindBundlerConfig.fromMetroConfig(config$1.uniwind, config.Platform.Web);
